src/Plugin/Field/FieldFormatter/FileImageFormatter.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\file_entity\Plugin\Field\FieldFormatter\FileImageFormatter.
+ */
+
+namespace Drupal\file_entity\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Field\EntityReferenceFieldItemListInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\image\Plugin\Field\FieldFormatter\ImageFormatter;
+
+/**
+ * Implementation of the 'image' formatter for the file_entity files.
+ *
+ * @FieldFormatter(
+ *   id = "file_image",
+ *   label = @Translation("File Image"),
+ *   field_types = {
+ *     "uri"
+ *   }
+ * )
+ */
+class FileImageFormatter extends ImageFormatter {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items) {
+    $elements = array();
+    $files = [$items->getEntity()];;
```

Comment:
I think we won't be able to use the parent implementation of this, so lets try to simplify it. Drop $files, there's just one, hardcode $delta 0, no empty check, no loop, just $file = $item->getEntity() and use that directly..